### PR TITLE
Improve checkout filters arguments description

### LINKED ISSUE #9385: Optimise `Available filters` documentation
Currently, the documentation of our `Available filters` is rather confusing. At the top of the page, we’re showing a description and a table with the parameters of the corresponding filter. At the bottom of the page, we’re showing code examples. Sometimes, we show a screenshot in the upper section, sometimes we show a screenshot in the lower section. I propose the following two changes:

1. Create one individual page per filter.
2. Structure each individual page as follows:
```
- Description
- Function Signatures
- Code Examples
- Screenshots
- Additional Resources
- Changelog (e.g. `__experimentalRegisterCheckoutFilters` had been graduaded to the stable in #8346)
```

Applying these suggestions to https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/checkout-block/available-filters.md, leads to the following file/folder structure:

```
available-filters.md
├── cart-line-items.md
├── order-summary-items.md
├── totals-footer-item-in-mini-cart-cart-and-checkout.md
├── coupons.md
├── checkout-and-place-order-button.md
└── additional-cart-checkout-inner-block-types.md
```

### LINKED ISSUE #10551: Improve the description of the arguments passed to checkout filters
In the [Available Filters](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/packages/checkout/filter-registry/README.md#available-filters) document, we note what arguments get passed to filter functions, however we do not give an in-depth description of these.

We should study the code and try to offer more information on each of the arguments passed to each filter (you may also wish to update https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/checkout-block/available-filters.md with the argument descriptions too).

For this ticket to be closed, a developer should be able to visit our documentation, look up any of the current filter and know exactly which arguments will be passed and what will be in them.

e.g. if they are going to register a callback for `cartItemPrice` they should be able to look at the documentation for that, see that their callback will receive: 
- `defaultValue: '<price/>'`,
- `extensions`
- `arg`
  - `context: 'cart'`
  - `cartItem: lineItem` (the current Cart Line Item being rendered),
  - `cart` (the current cart, will be equal to what's in the `wc/store/cart` data store),
  - `validation: productPriceValidation` (what is `productPriceValidation`? the developer should know what their value will be validated against)

### PR DESCRIPTION
**Pull request by nielslange**
## What

Improve the description of the arguments passed to the checkout filters.

Fixes #10551 
Fixes #9385

## Why

Currently, the arguments passed to the checkout filters are not documented properly. This makes it challenging for an external developer on how to use these filters correctly. This PR aims to improve the documentation of the arguments passed to the checkout filters.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Look up the *.md files of this PR.
2. Verify that all arguments are documented correctly.
3. Verify that the docs do not contain broken links.
4. Verify that all screenshots contain alt tags.
5. Verify that all footer links _"🐞 Found a mistake, or have a suggestion?"_ link to the corresponding docs.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

n/a

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [x] This PR has had any necessary documentation added/updated.

## Changelog

n/a

### CODE REVIEW

File: docs/third-party-developers/extensibility/checkout-block/available-filters/cart-line-items.md [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+registerCheckoutFilters( 'example-extension', {
+	cartItemClass: modifyCartItemClass,
+} );
+```
+
+> 💡 Filters can be also combined. See [Combined filters](../available-filters.md#combined-filters) for an example.
+
+### Screenshot <!-- omit in toc -->
+
+![Cart Item Class filter](https://woocommerce.com/wp-content/uploads/2023/10/Screenshot-2023-10-26-at-14.14.17.png)
```
Comment (tarhi-saad) [inline docs/third-party-developers/extensibility/checkout-block/available-filters/cart-line-items.md]: This image is broken.

**Review by tarhi-saad — CHANGES_REQUESTED** [review]
Thank you, @nielslange, for your outstanding efforts! Your dedication and attention to enhancing this documentation is truly commendable. Regarding the screenshots, pinpointing the changes can be challenging. Incorporating red arrows or circles to highlight specific areas could greatly guide developers. Additionally, zooming in on the focal points would be beneficial, as currently, we need to open the images in a new page to view details more clearly. Lastly, indicating which images represent 'before' and 'after' scenarios would significantly aid in quick comprehension. Here is an example from the doc before this PR:

| Before                                                                                                                                   | After                                                                                                                                    |
| ---------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
| image | image | 

And here is our new screenshot:

Show Remove Item Link filter

**Comment by nielslange** [comment]
Thanks for your feedback, @tarhi-saad. I'll take care of the changes.

**Comment by nielslange** [comment]
Hi @tarhi-saad, I've optimised all the screenshots. Would you mind doing another review?

**Review by tarhi-saad — APPROVED** [review]
Thank you, Niels, for the changes! The screenshots looks great! Let's 🚢

**Pull request merged**